### PR TITLE
Change from darwin to macos in editor

### PR DESCRIPTION
**Pull request by britzl**
We've recently changed platform name from x86_64-darwin to x86_64-macos in bob and build scripts but not in the editor.

### CODE REVIEW

**Review by JCash — DISMISSED** [review]
Thanks!

File: editor/src/clj/editor/updater.clj [hunk 1]
```diff
@@ -337,14 +337,14 @@
         install-dir (.getCanonicalFile
                       (if-let [path (system/defold-resourcespath)]
                         (case os
-                          "darwin" (io/file path "../../")
+                          "macos" (io/file path "../../")
```
Comment (JCash) [inline editor/src/clj/editor/updater.clj:340]: This is a part where I'd say that the editor doesn't need a copy of Platform.java

File: editor/src/clj/editor/util.clj [hunk 2]
```diff
@@ -133,14 +133,14 @@
        (.compare c (f a) (f b))))))
 
 (defn os-raw
-  "Returns :win32, :darwin or :linux"
+  "Returns :win32, :macos or :linux"
```
Comment (JCash) [inline editor/src/clj/editor/util.clj:136]: I think this type of editor code shouldn't need a duplicate of the PLatform.java.
But that's for the future tasks...

**Review by matgis — CHANGES_REQUESTED** [review]
You need to either edit or delete the editor's copy of `editor/src/java/com/defold/editor/Platform.java` as well, right?

**Review by matgis — DISMISSED** [review]
Oh, I didn't realize Mathias already changed it on `dev`. Carry on, I guess!

**Comment by britzl** [comment]
> This is a part where I'd say that the editor doesn't need a copy of Platform.java

> You need to either edit or delete the editor's copy of editor/src/java/com/defold/editor/Platform.java as well, right?

I did not even notice this! I assumed it used Platform.java from bob!

**Comment by britzl** [comment]
@JCash and @matgis I switched to use of Platform.java from bob instead of the old copy we had in the Editor

**Review by matgis — APPROVED** [review]
I think I've fixed the final editor issue. This should be good to merge now!

**Pull request merged**